src/Cluster.php

```diff
@@ -167,6 +167,33 @@ public function cleanDBonPurge()
         );
     }
 
+    /**
+     * Get the cluster of an item
+     *
+     * @param CommonDBTM $item
+     *
+     * @return Cluster|null
+     */
+    public static function getClusterByItem($item)
+    {
+        $itemCluster = new Item_Cluster();
+        if (
+            $itemCluster->getFromDBByCrit([
+                'itemtype' => $item->getType(),
+                'items_id' => $item->getID()
+            ])
+        ) {
+            if ($itemCluster->fields['clusters_id'] != 0) {
+                $cluster = new self();
+                $cluster->getFromDB($itemCluster->fields['clusters_id']);
+
+                return $cluster;
+            }
+        }
```

Comment:
Code could be more "defensive".
```suggestion
        $cluster = new self();
        $item_cluster = new Item_Cluster();
        if (
            $item_cluster->getFromDBByCrit(['itemtype' => $item->getType(), 'items_id' => $item->getID()])
            && $item_cluster->fields['clusters_id'] != 0
            && $cluster->getFromDB($item_cluster->fields['clusters_id'])
        ) {
            return $cluster;
        }
```